src/pip/_internal/network/lazy_wheel.py

```diff
@@ -0,0 +1,204 @@
+"""Lazy ZIP over HTTP"""
+
+__all__ = ['LazyZip']
+
+from bisect import bisect_left, bisect_right
+from contextlib import contextmanager
+from tempfile import NamedTemporaryFile
+from zipfile import BadZipfile, ZipFile
+
+from pip._vendor.requests.models import CONTENT_CHUNK_SIZE
+from pip._vendor.six.moves import range
+
+from pip._internal.network.utils import HEADERS, response_chunks
+from pip._internal.utils.typing import MYPY_CHECK_RUNNING
+
+if MYPY_CHECK_RUNNING:
+    from typing import Any, Dict, Iterator, List, Optional, Tuple
+
+    from pip._vendor.requests.models import Response
+
+    from pip._internal.network.session import PipSession
+
+
+class LazyZip:
```

Comment:
Do you think we should inherit from `IO` and make mypy happy, since [it would require many unused methods](https://github.com/McSinyx/lazip/blob/6309d68b7b5c054980eeac70c5b6941a5e83b35b/lazip.py#L36)?